civis/parallel.py

```diff
@@ -161,34 +166,34 @@ def infer_backend_factory(required_resources=None,
     # Update arguments with input
     state.arguments.update(arguments or {})
 
-    return make_backend_factory(docker_image_name=state.docker_image_name,
-                                docker_image_tag=state.docker_image_tag,
-                                repo_http_uri=state.repo_http_uri,
-                                repo_ref=state.repo_ref,
-                                required_resources=state.required_resources,
+    # Set defaults on other keyword arguments with
+    # values from the current script
+    for key in ['docker_image_name', 'docker_image_tag', 'repo_http_uri',
+                'repo_ref', 'remote_host_credential_id', 'git_credential_id',
+                'cancel_timeout', 'time_zone']:
+        kwargs.setdefault(key, state[key])
```

Comment:
That was the first thing I wanted to do. I see two problems -- first, not all keys in the state are valid input parameters. For example, there's a "last_run" and a "state". (And there's no obvious way to introspect valid input parameters. The API client uses **kwargs here.) Second, not all valid inputs are appropriate to transfer to child jobs. For example, I don't think we should reproduce the notification settings.